### PR TITLE
fix: grid2: quantity fee reduction for quote currency

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

File: pkg/strategy/grid2/strategy.go [hunk 1]
```diff
@@ -405,6 +427,11 @@ func (s *Strategy) processFilledOrder(o types.Order) {
 
 		// use the profit to buy more inventory in the grid
 		if s.Compound || s.EarnBase {
+			// if it's not using the platform fee currency, reduce the quote quantity for the buy order
+			if feeCurrency == s.Market.QuoteCurrency {
+				orderExecutedQuoteAmount = orderExecutedQuoteAmount.Sub(feeQuantityReduction)
```
Comment (gx578007) [inline pkg/strategy/grid2/strategy.go:432]: What if the fee currency is base currency?

File: pkg/strategy/grid2/strategy.go [hunk 2]
```diff
@@ -414,19 +441,7 @@ func (s *Strategy) processFilledOrder(o types.Order) {
...
-		s.logger.Infof("GRID BUY ORDER BASE FEE: %s %s", baseSellQuantityReduction.String(), s.Market.BaseCurrency)
-
-		baseSellQuantityReduction = roundUpMarketQuantity(s.Market, baseSellQuantityReduction)
-		s.logger.Infof("GRID BUY ORDER BASE FEE (Rounding with precision %d): %s %s",
-			s.Market.VolumePrecision,
-			baseSellQuantityReduction.String(),
-			s.Market.BaseCurrency)
-
-		newQuantity = newQuantity.Sub(baseSellQuantityReduction)
+		newQuantity = newQuantity.Sub(feeQuantityReduction)
```
Comment (gx578007) [inline pkg/strategy/grid2/strategy.go:446]: Need to check fee is in quote or base currency here?

Comment (c9s) [reply pkg/strategy/grid2/strategy.go:446]: You're right

File: pkg/strategy/grid2/strategy.go (hunk 1 above)
Comment (c9s) [reply pkg/strategy/grid2/strategy.go:434]: it's sell triggers buy order, so we only care about the quote currency here?

File: pkg/strategy/grid2/strategy.go (hunk 2 above)
Comment (c9s) [reply pkg/strategy/grid2/strategy.go]: Fixed

**Review by gx578007 — APPROVED** [review]

**Pull request merged**